aws/resource_aws_efs_access_point.go

```diff
@@ -0,0 +1,404 @@
+	conn := meta.(*AWSClient).efsconn
+	resp, err := conn.DescribeAccessPoints(&efs.DescribeAccessPointsInput{
+		AccessPointId: aws.String(d.Id()),
+	})
+	if err != nil {
+		if isAWSErr(err, efs.ErrCodeAccessPointNotFound, "") {
+			log.Printf("[WARN] EFS access point %q could not be found.", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading EFS access point %s: %s", d.Id(), err)
+	}
+
+	if hasEmptyAccessPoints(resp) {
+		return fmt.Errorf("EFS access point %q could not be found.", d.Id())
+	}
+
+	ap := resp.AccessPoints[0]
+
+	log.Printf("[DEBUG] Found EFS access point: %#v", ap)
+
+	d.SetId(*ap.AccessPointId)
+
+	fsARN := arn.ARN{
+		AccountID: meta.(*AWSClient).accountid,
+		Partition: meta.(*AWSClient).partition,
+		Region:    meta.(*AWSClient).region,
+		Resource:  fmt.Sprintf("file-system/%s", aws.StringValue(ap.FileSystemId)),
+		Service:   "elasticfilesystem",
+	}.String()
+
+	d.Set("file_system_arn", fsARN)
+	d.Set("file_system_id", ap.FileSystemId)
+	d.Set("arn", ap.AccessPointArn)
+	d.Set("owner_id", ap.OwnerId)
+
+	if err := d.Set("posix_user", flattenEfsAccessPointPosixUser(ap.PosixUser)); err != nil {
+		return fmt.Errorf("error setting posix user: %s", err)
+	}
+
+	if err := d.Set("root_directory", flattenEfsAccessPointRootDirectory(ap.RootDirectory)); err != nil {
+		return fmt.Errorf("error setting root directory: %s", err)
+	}
+
+	if err := d.Set("tags", keyvaluetags.EfsKeyValueTags(ap.Tags).IgnoreAws().Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsEfsAccessPointDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).efsconn
+
+	log.Printf("[DEBUG] Deleting EFS access point %q", d.Id())
+	_, err := conn.DeleteAccessPoint(&efs.DeleteAccessPointInput{
+		AccessPointId: aws.String(d.Id()),
+	})
+	if err != nil {
+		return err
+	}
+
+	err = waitForDeleteEfsAccessPoint(conn, d.Id(), 10*time.Minute)
+	if err != nil {
+		return fmt.Errorf("Error waiting for EFS access point (%q) to delete: %s", d.Id(), err.Error())
+	}
+
+	log.Printf("[DEBUG] EFS access point %q deleted.", d.Id())
+
+	return nil
+}
+
+func waitForDeleteEfsAccessPoint(conn *efs.EFS, id string, timeout time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{efs.LifeCycleStateAvailable, efs.LifeCycleStateDeleting, efs.LifeCycleStateDeleted},
+		Target:  []string{},
+		Refresh: func() (interface{}, string, error) {
+			resp, err := conn.DescribeAccessPoints(&efs.DescribeAccessPointsInput{
+				AccessPointId: aws.String(id),
+			})
+			if err != nil {
+				if isAWSErr(err, efs.ErrCodeAccessPointNotFound, "") {
+					return nil, "", nil
+				}
+
+				return nil, "error", err
+			}
+
+			if hasEmptyAccessPoints(resp) {
+				return nil, "", nil
+			}
+
+			mt := resp.AccessPoints[0]
+
+			log.Printf("[DEBUG] Current status of %q: %q", *mt.AccessPointId, *mt.LifeCycleState)
+			return mt, *mt.LifeCycleState, nil
```

Comment:
Same here regarding AWS Go SDK conversion functions:

```suggestion
			log.Printf("[DEBUG] Current status of %q: %q", aws.StringValue(mt.AccessPointId), aws.StringValue(mt.LifeCycleState))
			return mt, aws.StringValue(mt.LifeCycleState), nil
```